service/src/java/org/apache/hive/service/auth/ldap/DirSearch.java

```diff
@@ -34,6 +34,16 @@ public interface DirSearch extends Closeable {
    */
   String findUserDn(String user) throws NamingException;
 
+  /**
+   * Finds user's distinguished name.
+   * @param user username
+   * @param userSearchFilter Generic LDAP Search filter for ex: (&(uid={0})(objectClass=person))
+   * @param baseDn LDAP BaseDN for user searches for ex: dc=apache,dc=org
+   * @return DN for the specific user if exists, null otherwise
+   * @throws NamingException
+   */
+  String findUserDnBySearch(String user, String userSearchFilter, String baseDn) throws NamingException;
```

Comment:
Couldn't we reuse the 'findUserDn' method name (ie overload) for these new methods?

Reply:
There is a `DirSearch` class in the metastore package also. Should we introduce a similar change there also to keep the consistency? Not sure if LDAP search in metastore is used at all.

Reply:
yeah, this entire code was replicated for supporting ldap auth for HMS. I think it would make sense to make changes to the HMS provider as well. I wasnt sure how to test it manually though. Will give it a try otherwise may have to fork the work for another jira.

It is possible to lump them both into single method. I kept them separate for a couple reasons. findUserDn() and findUserDnBySearch() use different criteria/configuration to find the userDN from a given username.  This requires a change to the interface method though, which I wasn't very fond of. This also kept the methods separate based on the factory that was calling it. Less intersection with existing code. As this is an alternate configuration for LDAP.
if you feel strongly about merging them, I can give it a shot.

Reply:
I'm confused and missing the obvious;  I'm just suggesting renaming `String findUserDnBySearch(String user, String userSearchFilter, String baseDn) throws NamingException; `to `String findUserDn(String user, String userSearchFilter, String baseDn) throws NamingException;` . The new method with search arguments seem a nice extension (overload) to the original one.